src/sage/algebras/vertex_operators.py

```diff
@@ -0,0 +1,905 @@
+from sage.structure.sage_object import SageObject
+from sage.structure.unique_representation import UniqueRepresentation
+
+
+class FermionicFockSpace(CombinatorialFreeModule):
+    r"""
+    A model of the Fermionic Fock space `\mathcal{H}_F`. As a vector space, `\mathcal{H}_F`
+    has basis given by pairs `(\lambda, n)` where `\lambda` is a partition, and `n` is an integer.
+
+    EXAMPLES::
+
+        sage: from sage.algebras.vertex_operators import *
+        sage: F = FermionicFockSpace(QQ)
+        sage: F.an_element()
+        |[], 0> + 4*|[], 1> + 2*|[1], 0>
+    """
+    def __init__(self, R=QQ):
+        r"""
+        Initialize ``self``.
+
+        INPUT:
+
+        - ``R`` -- a ring (default: ``QQ``)
+
+        TESTS::
+
+            sage: from sage.algebras.vertex_operators import FermionicFockSpace
+            sage: F = FermionicFockSpace(QQ)
+            sage: TestSuite(F).run()
+        """
+        index_set = cartesian_product((Partitions(), ZZ))
+        CombinatorialFreeModule.__init__(self, R, index_set, prefix='', bracket='')
+
+    def _repr_term(self, m):
+        r"""
+        String representation of a basis element ``m`` of ``self``
+
+        EXAMPLES::
+
+            sage: from sage.algebras.vertex_operators import FermionicFockSpace
+            sage: F = FermionicFockSpace()
+            sage: print(F._repr_term(([3,2,1], -4)))
+            |[3, 2, 1], -4>
+        """
+        return '|' + str(m[0]) + ', ' + str(m[1]) + '>'
+
+
+def BosonicFockSpace(R=QQ, sym_basis='s', name='w'):
+    r"""
+    Bosonic Fock space.
+
+    INPUT:
+
+    - ``R`` -- (default: ``QQ``) a ring
+    - ``sym_basis`` -- (default: ``'s'``) string or realization of ``SymmetricFunctions(R)``
+    - ``name`` -- (default ``'w'``) string indicating the name of the Laurent variable.
+
+    OUTPUT:
+
+    The Laurent polynomial ring over the ring of symmertic functions over ``R`` in
+    the specified basis.
+
+    EXAMPLES::
+
+        sage: from sage.algebras.vertex_operators import BosonicFockSpace
+        sage: B = BosonicFockSpace(); B
+        Univariate Laurent Polynomial Ring in w over Symmetric Functions
+        over Rational Field in the Schur basis
+        sage: B.an_element()
+        s[]*w
+        sage: Sym = SymmetricFunctions(QQ); Sym.inject_shorthands(verbose=False)
+        sage: BosonicFockSpace(QQ, s)
+        Univariate Laurent Polynomial Ring in w over
+        Symmetric Functions over Rational Field in the Schur basis
+        sage: BosonicFockSpace(FractionField(QQ['q','t']), 'p')
+        Univariate Laurent Polynomial Ring in w over Symmetric Functions
+        over Fraction Field of Multivariate Polynomial Ring in q, t over
+        Rational Field in the powersum basis
+    """
+    Sym = SymmetricFunctions(R)
+    if isinstance(sym_basis, str):
+        b = getattr(Sym, sym_basis, None)
+        if b is None or b() not in Sym.realizations():
+            raise ValueError("Unknown symmetric function basis")
```

Comment:
```suggestion
            raise ValueError("unknown symmetric function basis")
```
We follow Python's conventions.